dotnet/src/Connectors/Connectors.AI.OpenAI/ChatCompletion/ChatHistoryExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+#pragma warning disable IDE0130
+using System.Collections.Generic;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.AI.ChatCompletion;
+#pragma warning restore IDE0130
+
+/// <summary>
+/// OpenAI-specific extensions to the <see cref="ChatHistory"/> class.
+/// </summary>
+public static class ChatHistoryExtensions
+{
+    /// <summary>
+    /// Add a function message to the chat history
+    /// </summary>
+    /// <param name="chatHistory">Chat history</param>
+    /// <param name="message">Message content</param>
+    /// <param name="functionName">Name of the function that generated the content</param>
+    public static void AddFunctionMessage(this ChatHistory chatHistory, string message, string functionName)
+    {
+        chatHistory.AddMessage(AuthorRole.Function, message, new Dictionary<string, string> { { "Name", functionName } });
+    }
+
+    /// <summary>
+    /// Add an assistant message to the chat history.
+    /// </summary>
+    /// <param name="chatHistory">Chat history</param>
+    /// <param name="chatResult">Chat result from the model</param>
+    public static void AddAssistantMessage(this ChatHistory chatHistory, IChatResult chatResult)
+    {
+        var chatMessage = chatResult.ModelResult.GetOpenAIChatResult().Choice.Message;
+
+        if (!chatMessage.Content.IsNullOrEmpty())
```

Comment:
Can a chat message have both text and a function call? The code suggests that it's possible.

Reply:
@SergeyMenshykh As you have quoted from OpenAI documentation,
```
messages.append(response_message)  # extend conversation with assistant's reply
```
I believe that we should simply extend the response_message from LLM, regardless of its content to the chat history. The response_message may contain message or function call or both because OpenAI's API reference does not explicitly mention that they cannot co-exist. You are correct that if they both exist, the code will generate two messages instead of one. Therefore, I suggest below changes:
```
public static void AddAssistantMessage(this ChatHistory chatHistory, IChatResult chatResult)
{
    var chatMessage = chatResult.ModelResult.GetOpenAIChatResult().Choice.Message;

    var functionCall = chatMessage.FunctionCall;
    if (functionCall is not null)
    {
        chatHistory.AddMessage(AuthorRole.Assistant, chatMessage.Content, new Dictionary<string, string>
        {
            { "Name", functionCall.Name },
            { "Arguments", functionCall.Arguments }
        });
    } else
    {
        chatHistory.AddAssistantMessage(chatMessage.Content);
    }
}
```